### DISCUSSION TITLE
How do I implement multiple roles like admin, teacher, and student in FilamentPHP to restrict access to specific panels?

### DISCUSSION THREAD

**Discussion by Pola5h** [filamentphp/filament#13728] [Help]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

I have four panels AccountsPanelProvider, AdminPanelprovider, StudentPanelProvider, and TeacherPanelProvider. Now I want to ensure that only the admin can access the admin panel, students can access the student panel, and so on for others. How can I do it? Can I implement it using like: `->authMiddleware([
                Authenticate::class,
                CheckAdmin::class,
            ]);`?

**Comment by dissto** [comment]
You can probably match the id of the panel in the user model...something like:

```php

public function canAccessPanel(Panel $panel): bool
{
  return match ($panel->getId()) {
      'admin' => $this->isAdmin(), // or whatever checks you have
      'student' => $this->isStudent(), // or whatever checks you have
      'teacher' => $this->isTeacher(), // or whatever checks you have
      default => false,
  };
}
```

See: https://filamentphp.com/docs/3.x/panels/users#authorizing-access-to-the-panel

**Reply by mhbb8897** [reply]
What if i want to restrict access for panel user in /user (after login) if i change the route locally with /admin they can access too? 
https://github.com/user-attachments/assets/bcf5f4b8-23af-4dcf-9170-f2c0e963104b
